### PR TITLE
Optimize immediate sub

### PR DESCRIPTION
**Pull request by hackaugusto**
## Describe your changes

Follows the same pattern as immediate addition and when subtracting by zero translates to a `Nop` instead of two opcodes, saving one cycle on this edge case.

## Checklist before requesting a review
- Repo forked and branch created from `next` according to naming convention.
- Commit messages and codestyle follow [conventions](./CONTRIBUTING.md).
- Relevant issues are linked in the PR description.
- Tests added for new functionality.
- Documentation/comments updated according to changes.

### CODE REVIEW

**Comment by hackaugusto** [comment]
I wasn't sure who request a review from, so I just used GH's suggestions. Since the team is growing, it would be nice to have [CODEOWNERS](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) to make that easy :)

File: assembly/src/assembler/instruction/field_ops.rs [hunk 1]
```diff
@@ -28,6 +28,18 @@ pub fn add_imm(span: &mut SpanBuilder, imm: Felt) -> Result<Option<CodeBlock>, A
     }
 }
 
+/// Appends a sequence of operations to subtract an immediate value to the value at the top of the
```
Comment (grjte) [inline assembly/src/assembler/instruction/field_ops.rs:31]: nit: "to subtract an immediate value from the value..."

**Comment by hackaugusto** [comment]
I wonder why we have:

```rust
    } else if imm == TWO {
        span.add_ops([Incr, Incr])  # this!
    } else {
        span.add_ops([Push(imm), Add])
```

The number of opcodes is the same, so it doesn't optimize the number of cycles. Do we have different costs for different opcodes other than the cycle count?

**Comment by grjte** [comment]
> 

Actually, yes! This comes down to [how we build span blocks](https://0xpolygonmiden.github.io/miden-vm/design/programs.html#span-block). Immediate values take up a full "group", whereas in the same space we can fit 9 instructions. It's an edge case, but more efficient to avoid using an immediate value here.

**Review by grjte — APPROVED** [review]
Looks good, thank you! One optional nit inline

File: assembly/src/assembler/instruction/field_ops.rs (hunk 1 above)
Comment (hackaugusto) [reply assembly/src/assembler/instruction/field_ops.rs]: done :)

**Review by grjte — APPROVED** [review]
Looks great, thanks!

**Pull request merged**